src/Application.tsx

```diff
@@ -1,35 +1,59 @@
-import React, { Suspense } from "react";
+import React, { Suspense, useEffect } from "react";
 import {
   Route,
-  Switch,
   RouteComponentProps,
+  Switch,
+  withRouter,
 } from "react-router-dom";
-import routes from "src/common/routes";
 
-const Application: React.FunctionComponent<{}> = (props) => {
+import { guestRoutes, userRoutes } from "src/common/routes";
+import { routes } from "./common/constants/pageRoutes";
+import { useAuth } from "./context/GlobalContext";
+
+const Application: React.FC<
+  RouteComponentProps<{}, {}, { from: { pathname: string } }>
+> = ({ history }) => {
+  const { state } = useAuth();
+
+  useEffect(() => {
+    // Store previous route in history state to return user back to that route
+    // from which it came from after user verified from auth listener.
+    if (state.user) {
+      const { state } = history.location;
+      history.replace(state?.from?.pathname ?? routes.HOME);
```

Comment:
Do we want to send them to `HOME` if `state?.from?.pathname` is not defined. I think it's fair to let the user land on the 404 page even if they are signed in.

Reply:
they will land on 404 page this case is for handling if previous route was also auth then user need to landon home page instead of auth page.